src/controller/AutoCommissioner.cpp

```diff
@@ -154,6 +154,18 @@ CHIP_ERROR AutoCommissioner::SetCommissioningParameters(const CommissioningParam
         mParams.SetSkipCommissioningComplete(params.GetSkipCommissioningComplete().Value());
     }
 
+    if (params.GetCountryCode().HasValue())
+    {
+        ChipLogProgress(Controller, "Setting country code from parameters");
+        mParams.SetCountryCode(params.GetCountryCode().Value());
+    }
+
+    if (params.GetDeviceRegulatoryLocation().HasValue())
+    {
+        ChipLogProgress(Controller, "Setting regulatory location type from parameters");
+        mParams.SetDeviceRegulatoryLocation(params.GetDeviceRegulatoryLocation().Value());
+    }
+
```

Comment:
Why is this needed, given the `mParams = params;` at the top of this function?

It seems there's a bunch of stuff in this method that should not be here; the only parts that really need to be are  the ones that change which buffers mParams points to.  All the things that get copied by that assignment should just be removed (in a separate PR).

Reply:
I created https://github.com/project-chip/connectedhomeip/pull/24422 to remove the unnecessary parts here.

Reply:
I removed the get then set here. I think your PR should work, at least it seemed to work when I pulled and tested, but it was not an exhaustive test.